ExtCore/Pervasive.fs

```diff
@@ -1267,65 +1230,53 @@ module Choice =
     /// <param name="state"></param>
     /// <returns></returns>
     [<CompiledName("FoldBack")>]
-    let foldBack (folder : 'T -> 'State -> 'State) (value : Choice<'T, 'Error>) (state : 'State) : 'State =
-        // Preconditions
-        checkNonNull "value" value
-
+    let foldBack (folder : 'T -> 'State -> 'State) (value : Result<'T, 'Error>) (state : 'State) : 'State =
         match value with
-        | Choice1Of2 result ->
+        | Ok result ->
             folder result state
-        | Choice2Of2 _ ->
+        | Error _ ->
             state
 
     /// <summary></summary>
     /// <param name="action"></param>
     /// <param name="value"></param>
     /// <returns></returns>
     [<CompiledName("Iterate")>]
-    let iter (action : 'T -> unit) (value : Choice<'T, 'Error>) : unit =
-        // Preconditions
-        checkNonNull "value" value
-
+    let iter (action : 'T -> unit) (value : Result<'T, 'Error>) : unit =
         match value with
-        | Choice2Of2 _ -> ()
-        | Choice1Of2 result ->
+        | Error _ -> ()
+        | Ok result ->
             action result
 
     /// <summary></summary>
     /// <param name="value"></param>
     /// <returns></returns>
     [<CompiledName("BindOrRaise")>]
-    let inline bindOrRaise (value : Choice<'T, #exn>) : 'T =
-        // Preconditions
-        checkNonNull "value" value
-
+    let inline bindOrRaise (value : Result<'T, #exn>) : 'T =
         match value with
-        | Choice1Of2 result ->
+        | Ok result ->
             result
-        | Choice2Of2 ex ->
+        | Error ex ->
             raise ex
 
     /// <summary></summary>
     /// <param name="value"></param>
     /// <returns></returns>
     [<CompiledName("BindOrFail")>]
-    let inline bindOrFail (value : Choice<'T, string>) : 'T =
-        // Preconditions
-        checkNonNull "value" value
-
+    let inline bindOrFail (value : Result<'T, string>) : 'T =
```

Comment:
Maybe `bindOrError`?

Reply:
Well, in this case it does a `fail` so it does not really return an error but raises an exception.

Reply:
Ah, sorry :)

Reply:
No worries :)